src/main/java/taxi/dao/impl/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,58 @@
+package taxi.dao.impl;
+
+import java.util.List;
+import java.util.Optional;
+import taxi.dao.ManufacturerDao;
+import taxi.db.Storage;
+import taxi.exception.NoSuchManufacturerException;
+import taxi.lib.Dao;
+import taxi.model.Manufacturer;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        Storage.addManufacturer(manufacturer);
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        return Storage.manufacturers
+                .stream()
+                .filter(man -> man.getId().equals(id))
+                .findAny();
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        return Storage.manufacturers;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        Optional<Manufacturer> targetId = get(manufacturer.getId());
+        if (targetId.isEmpty()) {
```

Comment:
as i know, you can use isPresent() into Optional

Reply:
or when you will return, just call the method orElseThrow()

Reply:
You are absolutely right about `isPresent` - but I wanted to `throw` at the very first step and then proceed if `targetId` is not empty.

Reply:
I have rewritten with `isPresent` - I'll give it a try)